opentracing-shim/src/main/java/io/opentelemetry/opentracingshim/SpanShim.java

```diff
@@ -214,10 +235,39 @@ static Attributes convertToAttributes(Map<String, ?> fields) {
       } else if (value instanceof Boolean) {
         attributesBuilder.put(booleanKey(key), (Boolean) value);
       } else {
-        attributesBuilder.put(stringKey(key), value.toString());
+        if (isExceptionEvent) {
+          switch (key) {
+            case Fields.ERROR_KIND:
+              key = SemanticAttributes.EXCEPTION_TYPE.getKey();
+              break;
+            case Fields.MESSAGE:
+              key = SemanticAttributes.EXCEPTION_MESSAGE.getKey();
+              break;
+            case Fields.STACK:
+              key = SemanticAttributes.EXCEPTION_STACKTRACE.getKey();
+              break;
+            default:
+              break;
+          }
+        }
+        if (!isErrorEvent || !key.equals(Fields.ERROR_OBJECT) || !(value instanceof Throwable)) {
+          attributesBuilder.put(stringKey(key), value.toString());
```

Comment:
This doesn't solve the problem of extra allocations. You're still allocating a new key here, when you already have the key above for the given cases. Also, updating the variable you're switching on inside the switch block is pretty sketchy. Please refactor this code to have another variable for the `AttributeKey` instance that will be used here, so you can get rid of the extra allocations, and make the code cleaner.

Reply:
refactored